src/CMakeLists.txt

```diff
@@ -9,7 +9,7 @@ include(${UMF_CMAKE_SOURCE_DIR}/cmake/helpers.cmake)
 # TODO: Cleanup the compile definitions across all the CMake files
 set(UMF_COMMON_COMPILE_DEFINITIONS UMF_VERSION=${UMF_VERSION})
 
-if(UMF_BUILD_LEVEL_ZERO_PROVIDER)
```

Comment:
actually I can see we set here `LEVEL_ZERO_INCLUDE_DIRS` and UR is setting `LEVEL_ZERO_INCLUDE_DIR` which may result in error in UR (after we'd disable that new option).

Perhaps we could:
- rename `LEVEL_ZERO_INCLUDE_DIRS` in this repo to `LEVEL_ZERO_INCLUDE_DIR`
- check here (instead of this new option) if `LEVEL_ZERO_INCLUDE_DIR` is set
- set UMF target dependent of level_zero adapter, or more precisely of the `level_zero_loader` target (if that's possible)

Reply:
We could just link with `LevelZeroLoader-Headers` target that is provided by UR (or create that target ourselves if UMF_FETCH_LEVEL_ZERO is set.

Reply:
I agree that we should add an option that disables fetching L0 headers but for providing existing ones to UMF I would rather use CMAKE_C_FLAGS and not introduce any UR-specific code that makes our cmake files more complex